data/src/Cardano/Data/StrictMaybe.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE EmptyDataDecls #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{- | Strict version of the 'Maybe' type..
+-}
+module Cardano.Data.StrictMaybe
+  ( StrictMaybe (SNothing,SJust),
+    -- * Conversion: StrictMaybe <--> Maybe
+    strictMaybeToMaybe,
+    maybeToStrictMaybe,
+    -- * accessing the optional component
+    fromSMaybe,
+  )
+  where
+
+import Data.Aeson (FromJSON (..), ToJSON (..))
+import NoThunks.Class (NoThunks (..))
+import GHC.Generics (Generic)
+import Control.DeepSeq (NFData)
+import Cardano.Binary
+  ( FromCBOR (fromCBOR),
+    ToCBOR (toCBOR),
+    decodeBreakOr,
+    decodeListLenOrIndef,
+    encodeListLen,
+  )
+import qualified Control.Monad.Fail
+import Data.Default.Class(Default(..))
+
+
+-- ===============================================
+
+data StrictMaybe a
+  = SNothing
+  | SJust !a
+  deriving (Eq, Ord, Show, Generic, Functor, Foldable, Traversable)
+
+instance NoThunks a => NoThunks (StrictMaybe a)
+
+instance NFData a => NFData (StrictMaybe a)
```

Comment:
I think you can add these two to the `deriving` clause above (thanks to `DerivingAnyClass`).

Reply:
done